engine.py

```diff
@@ -130,43 +131,70 @@ def validate_context(self, tank, context):
         '''
         if context.project:
             if not context.entity:
-                task = self.select_task_ui(context)
-                newContext = tank.context_from_entity('Task', task['id'])
+                newContext = self.userChosenContext(tank, context)
                 return newContext
             elif context.entity and not context.task:
-                filters = [
-                    ['project', 'is', context.project],
-                    ['entity', 'is', context.entity],
-                ]
-                task = None
-                if context.entity['type'] == 'Shot':
-                    filters.append(['content', 'is', 'Lighting'])
-                    task = tank.shotgun.find_one("Task", filters)
-                elif context.entity['type'] == 'Asset':
-                    filters.append(['content', 'is', 'Shading'])
-                    task = tank.shotgun.find_one("Task", filters)
+                task = self.getTask(tank, context)
                 if task:
                     newContext = tank.context_from_entity('Task', task['id'])
                     return newContext
                 else:
-                    task = self.select_task_ui(context)
-                    newContext = tank.context_from_entity('Task', task['id'])
+                    newContext = self.userChosenContext(tank, context)
                     return newContext
             else:
                 return context
         return context
 
-    def select_task_ui(self, context):
-        taskNames = ['Lighting', 'Shading', 'FX']
-        shotPreferredTask = 'Lighting'
-        assetPreferredTask = 'Shading'
-        tc = taskChooser.TaskChooser(context, taskNames, shotPreferredTask, assetPreferredTask)
+    def getTask(self, tank, context):
```

Comment:
Your are redefining the `tank` package. And I haven't noticed in the previous PR, but ou are also redefining it in `validate_context`. This should be fixed to avoid any weird behavior I think.